### DISCUSSION TITLE
faq/env/windows/

### DISCUSSION THREAD

**Discussion by giscus** [physics-data/faq#181] [Giscus]
# faq/env/windows/

FAQ for Big Data in Experimental Physics

https://physics-data.meow.plus/faq/env/windows/

**Comment by 6yq** [comment]
好像有同学提出 WSL2 不能用 ```echo``` 写入，提供一种使用编辑器将环境变量 ```DISPLAY``` 写入到 ```~/.bashrc``` 中的具体方法，以vim为例：

打开 ```~/.bashrc``` 文档：

```objc
vim ~/.bashrc
```

```Ctrl+G``` 可以跳转到最后一行，按下 ```I``` 键进入插入模式，跳到最后一个字母处后回车另起一行，输入环境变量配置语句：

```objc
export DISPLAY=$(ip route show default | cut -d' ' -f3):0
```

**配置语句请以上文为准**

键入完成后按 ```esc``` 就可以退出插入模式。输入 ```:wq``` 并回车，可以关闭 ```~/.bashrc```。

_这时候输入的 ```:wq``` 应该是看不见的，请不用担心~_

上述过程可以使用不同的编辑器来完成，都只需要在末尾另起一行加入配置语句并保存。

更改完以后为了使改动立刻生效，而不用重新登录，需要使用 ```source``` _（这一步应该不管使用什么编辑器都需要输入，因为这是Linux本身更新的命令）_：

```objc
source ~/.bashrc
```

更改完成以后就可以 ```echo $DISPLAY``` 来检查有没有生效。

**Reply by yfzhao20** [reply]
> `export DISPLAY=$(ip route show default | cut -d' ' -f3):0`

试验了一下，事实上 WSL2 是可以使用 echo 写入的，但是简单使用 `echo 'export DISPLAY=$(ip route show default | cut -d' ' -f3):0' >> ~/.bashrc` 会因为中间两个单引号的转义问题，而在写入时把这两个单引号吃掉，因此可以尝试：
``` bash
echo "export DISPLAY=\$(ip route show default | cut -d' ' -f3):0" >> ~/.bashrc
```

这样似乎就可以写入了
